src/blockstorage/blockstorage.cpp

```diff
@@ -517,6 +517,8 @@ bool WriteUndoToDisk(const CBlockUndo &blockundo,
  */
 bool ReadUndoFromDisk(CBlockUndo &blockundo, const CDiskBlockPos &pos, const CBlockIndex *pindex)
 {
+    if (pindex == nullptr)
+        return error("Genesis block has no undo information");
```

Comment:
Even though we always check pindex before passing it into this function, this is a good check. However, the genesis block does have undo data. when pindex is nullptr it is the block before genesis block so this error message should be changed.

Reply:
for clarity, if you were to consider the undo data its own chain an, its "tip" would always be the blockchain tip - 1